pkg/vdr/web/did.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package web
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+)
+
+const (
+	defaultPath  = "/.well-known/doc.json"
+	documentPath = "/doc.json"
+)
+
+// parseDIDWeb consumes a did:web identifier and returns the URL location of the did Doc.
+func parseDIDWeb(id string) (string, error) {
+	var urlString string
+
+	parsedDID, err := did.Parse(id)
+	if err != nil {
+		return urlString, fmt.Errorf("invalid did, does not conform to generic did standard --> %w", err)
+	}
+
+	pathComponents := strings.Split(parsedDID.MethodSpecificID, ":")
+
+	if len(pathComponents) == 1 {
+		urlString = "https://" + pathComponents[0] + defaultPath
+		return urlString, nil
+	}
+
+	switch pathComponents[0] {
+	case "localhost", "127.0.0.1":
```

Comment:
Is this code added just for testing purposes? If so you should elaborate why because we usually don't add extra code to support testing.
Is this valid scenario for "web" VDR?

Reply:
yes it's just for testing, web dids shouldn't be originating from any site with a host:port url ~really~ as any organization trusted enough to have did:web would likely have a proper url

Reply:
from the spec: "The method specific identifier MUST match the common name used in the SSL/TLS certificate, and it MUST NOT include IP addresses or port numbers. Directories and subdirectories MAY optionally be included, delimited by colons rather than slashes. " https://w3c-ccg.github.io/did-method-web/#method-specific-identifier

Reply:
``
	switch pathComponents[0] {
	// This is for testing purposes ONLY and should not be used in production
	case "localhost", "127.0.0.1":
		urlString = "https://" + strings.Join(pathComponents[0:2], ":") + "/" + strings.Join(pathComponents[2:], "/")

		log.New("aries-framework/pkg/vdr/web").Warnf("using an ip address or port number does not conform to spec")
``

would something like this be acceptable ? @sandrask

Reply:
This is not acceptable.

Yes, `127.0.0.1` is not allowed by the spec, but `localhost` is.

Also, ports will be supported. There is a proposed solution to base64url-encode the host portion: https://github.com/w3c-ccg/did-method-web/issues/7#issuecomment-623485232. At least one implementation has adopted the proposal: https://github.com/ProjectProvenance/proof-points/pull/42/files.

@amodkala let's adopt the proposal - assume ports are allowed and the host:port portion is base64url-encoded.

Reply:
@amodkala my apologies - somehow I mixed up url-encoding with base64url-encoding this morning. The proposal for the `did:web` ID to have the `host:port` portion and all path components url-encoded. See the issue I linked to above for more details.

Reply:
Just realized url-encoding the path components makes it non-compliant with the generic did-core syntax

Reply:
Followup: #2336